test-framework/common/src/main/java/io/quarkus/test/common/http/TestHTTPResourceManager.java

```diff
@@ -102,7 +101,7 @@ public static void inject(Object testCase, List<Function<Class<?>, String>> endp
                     }
                     f.setAccessible(true);
                     try {
-                        f.set(testCase, provider.provide(new URI(val), f));
+                        f.set(testCase, provider.provide(val, f));
```

Comment:
Won't this break tests expecting the `URI` type?

Reply:
We have plenty of tests that use URI and they work (?) So it shouldn't be an issue